libraries/anvill_passes/src/TransformRemillJumpIntrinsics.cpp

```diff
@@ -202,17 +204,17 @@ bool TransformRemillJumpIntrinsics::runOnFunction(llvm::Function &func) {
 
     // Run private function passes if the jump instrinsics
     // are replaced
-    llvm::legacy::FunctionPassManager fpm(module);
-    fpm.add(llvm::createDeadCodeEliminationPass());
-    fpm.add(llvm::createSROAPass());
-    fpm.add(llvm::createCFGSimplificationPass());
-    fpm.add(llvm::createInstructionCombiningPass());
-    fpm.doInitialization();
-    fpm.run(func);
-    fpm.doFinalization();
+    llvm::FunctionPassManager fpm;
+    fpm.addPass(llvm::DCEPass());
+    fpm.addPass(llvm::SROA());
+    fpm.addPass(llvm::SimplifyCFGPass());
+    // TODO(alex): It doesn't seem to like this and complains about the pass not being registered.
+    // Perhaps it's because we're nesting pass managers?
+    // fpm.addPass(llvm::InstCombinePass());
```

Comment:
This thing where we do another series of passes in a nested way seems to be causing a bit of trouble.